odoo/tools/cache.py

```diff
@@ -63,8 +63,12 @@ def __call__(self, method):
     def determine_key(self):
         """ Determine the function that computes a cache key from arguments. """
         if self.skiparg is None:
+            # Remove annotation because lambda doesn't accept it
+            args = ', '.join(
+                str(params.replace(annotation=Parameter.empty))
+                for params in signature(self.method).parameters.values()
+            )
```

Comment:
It still need to remove the annotation with the `params.replace`. Support `*args` and `**kwargs` is still a nice to have. 

I move this commit in a other branch (because it is not related to language) : https://github.com/odoo/odoo/pull/109777